hera_cal/lstbin.py

```diff
@@ -143,6 +158,32 @@ def lst_bin(data_list, lst_list, flags_list=None, dlst=None, lst_start=None, lst
         # update all_lst_indices
         all_lst_indices.update(set(grid_indices[data_in_bin]))
 
+        if rephase:
+            # rephase each integration in d to nearest LST bin
+            if freq_array is None or antpos is None:
+                raise ValueError("freq_array and antpos is needed for rephase")
+
+            # check if d has already been phased (used for lst_bin_files)
+            try:
+                if hasattr(d, 'phase_type'):
+                    if d.phase_type == "drift":
+                        pass
+                    else:
+                        raise LSTBINPHASED
+
+                # copy rephased data if desired
+                if copy_rephase:
+                    d = copy.deepcopy(d)
+
+                # form baseline dictionary and lst_rephase
+                bls = odict(map(lambda k: (k, antpos[k[0]] - antpos[k[1]]), d.keys()))
+                lst_shift = lst_grid[grid_indices] - l
+                lst_rephase(d, bls, freq_array, lst_shift, lat=lat)
+                d.phase_type = "LSTBIN"
```

Comment:
d is a DataContainer right? Do we want to explicitly have a phase attribute on data containers? I'm all for it considering we're going to have multiple types of phased data products now.

Reply:
well, it gets a little tricky b/c if we have a `phase` attribute we also need a `phase_center` and `phase_epoch` as well, and the idea behind the `lst_rephase` function is that this can be different *for every single integration*, so this turns out to be a decent amount of data you need to carry around. This attribute is used only internally in the context of this single function, so I don't think its something we need to make part of the `DataContainer` object.